plaso/parsers/amcache.py

```diff
@@ -1,331 +1,324 @@
+    if modification_time_value:
+      modification_time = filetime.Filetime(
+          modification_time_value.get_data_as_integer())
+      event = time_events.DateTimeValuesEvent(
+          modification_time, definitions.TIME_DESCRIPTION_MODIFICATION)
+      parser_mediator.ProduceEventWithEventData(event, event_data)
+
+    compilation_time_value = file_reference_key.get_value_by_name(
+        self._AMCACHE_COMPILATION_TIME)
+    if compilation_time_value:
+      link_time = posix_time.PosixTime(
+          compilation_time_value.get_data_as_integer())
+      event = time_events.DateTimeValuesEvent(
+          link_time, definitions.TIME_DESCRIPTION_CHANGE)
+      parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def _ParseProgramKey(self, parser_mediator, program_key):
+    """Parses a program key (a sub key of Root\\Programs) for events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      program_key (pyregf_key): program key.
+    """
+    event_data = AMCacheProgramEventData()
+
+    for value_name, attribute_name in self._PRODUCT_KEY_VALUES.items():
+      value = program_key.get_value_by_name(value_name)
+      if not value:
+        continue
+
+      value_data = self._GetValueDataAsObject(parser_mediator, value)
+      setattr(event_data, attribute_name, value_data)
+
+    installation_time_value = program_key.get_value_by_name(
+        self._AMCACHE_P_INSTALLATION_TIME)
+    if installation_time_value:
+      installation_time = posix_time.PosixTime(
+          installation_time_value.get_data_as_integer())
       event = time_events.DateTimeValuesEvent(
-          posix_time.PosixTime(event_data.linkerts),
-          definitions.TIME_DESCRIPTION_CHANGE)
+          installation_time, definitions.TIME_DESCRIPTION_INSTALLATION)
       parser_mediator.ProduceEventWithEventData(event, event_data)
 
-manager.ParsersManager.RegisterParser(AmcacheParser)
+  def _ParseProgramsKey(self, parser_mediator, programs_key):
+    """Parses a Root\\Programs key.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      programs_key (pyregf.key): the Programs key.
+    """
+    for program_key in programs_key.sub_keys:
+      self._ParseProgramKey(parser_mediator, program_key)
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses an AMCache.hve file-like object for events.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_object (dfvfs.FileIO): file-like object.
+    """
+    regf_file = pyregf.file()
+    try:
+      regf_file.open_file_object(file_object)
+    except IOError:
+      # The error is currently ignored -> see TODO above related to the
```

Comment:
Please log the failure here at debug at least.

Reply:
the reason for not adding a debug print is that amcache.ghve are regf files that are also handled by the Registry parsing in plaso and this error will show up there as well.